### PR TITLE
refactor!: use azurerm_redis_cache resource

### PR DESCRIPTION
**Pull request by dougbw**
I have updated this module to use the native terraform azurerm_redis_cache resource instead of the arm template deployment.

(as per https://github.com/hmcts/cnp-module-redis/pull/19 but I have kept this on tf 0.11 syntax)

Notes for existing users of this module:
- Existing configuration blocks for invoking this module will remain valid as the required inputs have not changed
- Existing redis cache resources will not get deleted / re-provisioned through this change - only the arm deployment record itself is removed from the resource group
- Existing redis cache resources will need to be imported or will result a resource conflict. Example import command:
`terraform import module.redis-cache.azurerm_redis_cache.redis /subscriptions/bf308a5c-0624-4334-8ff8-8dca9fd43783/resourceGroups/core-infra-cache-idam-sandbox/providers/Microsoft.Cache/Redis/core-infra-idam-sandbox`

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -4,23 +4,26 @@ resource "azurerm_resource_group" "cache-resourcegroup" {
 
   tags = "${merge(var.common_tags,
     map("lastUpdated", "${timestamp()}")
-    )}"
+  )}"
```
Comment (timja) [inline main.tf:7]: shouldn't the arm template be removed as well?

**Comment by timja** [comment]
does it need importing for azurerm 1.x users or will it safely upsert and modify the state file?

**Comment by dougbw** [comment]
existing users of the module would need to do an import if this change is implemented, terraform will not automatically import the redis resource if it already exists.

**Comment by timja** [comment]
> existing users of the module would need to do an import if this change is implemented, terraform will not automatically import the redis resource if it already exists.

normally it does work, and would be worth testing by creating a redis instance with master, and then running this branch over top of it.

that would allow us to merge this easier

File: main.tf [hunk 2]
```diff
@@ -4,23 +4,26 @@ resource "azurerm_resource_group" "cache-resourcegroup" {
 
   tags = "${merge(var.common_tags,
     map("lastUpdated", "${timestamp()}")
```
Comment (naikajah) [inline main.tf:6]: we should remove the lastUpdated tag from the code. Its not required

Comment (dougbw) [reply main.tf]: done

**Pull request merged**